refl1d/probe.py

```diff
@@ -1416,7 +1414,8 @@ def restore_data(self):
     restore_data.__doc__ = Probe.restore_data.__doc__
 
     def simulate_data(self, theory, noise=2):
-        if numpy.isscalar(noise):
+        noise = numpy.asarray(noise)
+        if len(noise.shape) < 2:
```

Comment:
Further: need to explicitly test for None.  Otherwise, `np.asarray(None)` => `array(None, dtype=object)`, which returns false for `(noise is None)`.

Reply:
Those are good points. There's not reason to convert to np arrays other than as a convenience to check the dimensions.

It's true that the different possibilities for the input dimensions create a little bit of a headache. I would almost vote for not dealing with four-tuples and use the same noise (array or scalar) for all cross-sections. That's likely the way it will be used. But check the extended condition I just added as a possible solution that allows all the scenarios. Note that we're not complaining if the dimensions are not compatible with the data... I didn't add it because it wasn't there before. We might want to add the extra check to verify that the length of dR is the length of Q.